python_modules/dagster-pipes/dagster_pipes/__init__.py

```diff
@@ -84,6 +84,10 @@ class PipesOpenedData(TypedDict):
     """
 
     extras: Mapping[str, Any]
+    log_writer: Optional["PipesLogWriterOpenedData"]
```

Comment:
Until we decide to introduce `log_writer` as part of the top-level APIs, I think we should avoid adding `log_writer` as a top-level key here. Any info can just be passed under `extras`, i.e. instead of passing under `log_writer.extras` (the current setup) just pass under `extras.log_writer`.